pkg/runner/local_runner.go

```diff
@@ -66,14 +69,60 @@ func convertTimeval(t syscall.Timeval) *duration.Duration {
 	}
 }
 
+func findExecutable(file string) error {
+	d, err := os.Lstat(file)
+	if err != nil {
+		return err
+	}
+	if m := d.Mode(); !m.IsDir() && m&0111 != 0 {
+		return nil
+	}
+	return os.ErrPermission
+}
+
+func (r *localRunner) lookPathChroot(file, chroot string) (string, error) {
+	if strings.Contains(file, "/") {
+		if err := findExecutable(file); err != nil {
+			return file, err
+		}
+		return file, nil
+	}
+	path := os.Getenv("PATH")
```

Comment:
I don't think this should use `os.Getenv()` to obtain `$PATH`. Instead, it should extract it from the environment stored in `RunRequest`.

The nice thing about a feature like this is that it can easily be implemented as a new decorator type for `Runner`. Think of it as a simple mutator for requests that come in: You get a `RunRequest` containing an `argv[0]` without a slash in it. You then try to resolve it to one that does, based on the `PATH` specified in its environment.

By modelling it that way, it becomes a lot easier to unit test this functionality.